### PR TITLE
Copy popover hover trigger service from #1840

### PR DESCRIPTION
**Pull request by Christian862**
## Description
The purpose of this PR is the break up https://github.com/hypertrace/hypertrace-ui/pull/1840 in to two PRs, with this PR being responsible only for the popover hover trigger service. This is because other work is dependent on this service.

Note: The changes in here are a direct copy and paste from https://github.com/hypertrace/hypertrace-ui/pull/1840 , which has undergone a few cycles of review. 

The service enables consumers to open and close popover content based `mouseenter` and `mouseleave` events (or others), and enables the popover to have actionable content (click, hover, scroll, etc.)

### Testing
Please describe the tests that you ran to verify your changes. Please summarize what did you test and what needs to be tested e.g. deployed and tested helm chart locally. 

### Checklist:
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### Documentation
Make sure that you have documented corresponding changes in this repository or [hypertrace docs repo](https://github.com/hypertrace/hypertrace-docs-website) if required.

### CODE REVIEW

**Comment by itssharmasandeep** [comment]
Sorry, I know I didn't write tests in my PR :P, can we write it here, please?

**Comment by Christian862** [comment]
> Sorry, I know I didn't write tests in my PR :P, can we write it here, please?

haha, added some

**Comment by itssharmasandeep** [comment]
LGTM. But will for @anandtiwary to review it once

File: projects/components/src/public-api.ts [hunk 1]
```diff
@@ -266,6 +266,8 @@ export * from './popover/popover-ref';
 export * from './popover/popover.module';
 export * from './popover/popover.component';
 export * from './popover/popover.service';
+export * from './popover/popover-content.component';
```
Comment (itssharmasandeep) [inline projects/components/src/public-api.ts:269]: I don't think, we need to export it now. Do we?

Comment (Christian862) [reply projects/components/src/public-api.ts:269]: no we shouldn't need to, removing

**Review by itssharmasandeep — APPROVED** [review]

File: projects/components/src/popover/service/popover-hover-trigger.service.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
+import { ElementRef, Injectable, OnDestroy, TemplateRef } from '@angular/core';
+import { isNil } from 'lodash-es';
+import { EMPTY, Observable, Subject, Subscription } from 'rxjs';
+import { debounceTime, switchMap } from 'rxjs/operators';
+import { PopoverBackdrop, PopoverPositionType, PopoverRelativePositionLocation } from '../popover';
+import { PopoverRef } from '../popover-ref';
+import { PopoverService } from '../popover.service';
+
+@Injectable()
+export class PopoverHoverTriggerService implements OnDestroy {
```
Comment (anandtiwary) [inline projects/components/src/popover/service/popover-hover-trigger.service.ts:10]: @itssharmasandeep Do we have any plans to merge your tooltip changes?

**Pull request merged**